crates/swc_ecma_minifier/src/compress/optimize/unused.rs

```diff
@@ -164,6 +164,53 @@ impl Optimizer<'_> {
         params.retain(|p| !p.is_invalid());
     }
 
+    #[cfg_attr(feature = "debug", tracing::instrument(skip_all))]
+    fn take_ident_of_obj_pat_if_unused_and_side_effect_free(
```

Comment:
I think this is unnecessarily complex.

See https://github.com/swc-project/swc/pull/9478/files#diff-0ff76c3ab1246166f0a7132a4da5346082038c9dd821ab81d95343aa31510aa0R399-R408

Reply:
OK.
This pr could be closed and then follow #9478.